src/wp-includes/option.php

```diff
@@ -1413,64 +1395,37 @@ function get_network_option( $network_id, $option, $default = false ) {
 		return $pre;
 	}
 
-	// Prevent non-existent options from triggering multiple queries.
-	$notoptions_key = "$network_id:notoptions";
-	$notoptions     = wp_cache_get( $notoptions_key, 'site-options' );
-
-	if ( is_array( $notoptions ) && isset( $notoptions[ $option ] ) ) {
-
-		/**
-		 * Filters a specific default network option.
-		 *
-		 * The dynamic portion of the hook name, `$option`, refers to the option name.
-		 *
-		 * @since 3.4.0
-		 * @since 4.4.0 The `$option` parameter was added.
-		 * @since 4.7.0 The `$network_id` parameter was added.
-		 *
-		 * @param mixed  $default    The value to return if the site option does not exist
-		 *                           in the database.
-		 * @param string $option     Option name.
-		 * @param int    $network_id ID of the network.
-		 */
-		return apply_filters( "default_site_option_{$option}", $default, $option, $network_id );
-	}
-
 	if ( ! is_multisite() ) {
 		/** This filter is documented in wp-includes/option.php */
 		$default = apply_filters( 'default_site_option_' . $option, $default, $option, $network_id );
 		$value   = get_option( $option, $default );
 	} else {
-		$cache_key = "$network_id:$option";
-		$value     = wp_cache_get( $cache_key, 'site-options' );
-
-		if ( ! isset( $value ) || false === $value ) {
-			$row = $wpdb->get_row( $wpdb->prepare( "SELECT meta_value FROM $wpdb->sitemeta WHERE meta_key = %s AND site_id = %d", $option, $network_id ) );
-
-			// Has to be get_row() instead of get_var() because of funkiness with 0, false, null values.
-			if ( is_object( $row ) ) {
-				$value = $row->meta_value;
-				$value = maybe_unserialize( $value );
-				wp_cache_set( $cache_key, $value, 'site-options' );
-			} else {
-				if ( ! is_array( $notoptions ) ) {
-					$notoptions = array();
-				}
-
-				$notoptions[ $option ] = true;
-				wp_cache_set( $notoptions_key, $notoptions, 'site-options' );
+		$meta = get_metadata_raw( 'site', $network_id, $option );
+		if ( is_array( $meta ) && ! empty( $meta ) ) {
+			$value = array_shift( $meta );
+		} else {
+			/**
```

Comment:
Can we move the hook documentation up to the first occurrence above. I thought that the standard was just the first occurrence, but [looks like it's actually](https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#4-1-duplicate-hooks) to "use [svn blame](https://make.wordpress.org/core/handbook/svn/code-history/#using-subversion-annotate) to find the first use of the hook in terms of the earliest revision", which would now be the one within the `if ( ! is_multisite() )` check.

Reply:
Fixed